doc/protocol.txt

```diff
@@ -400,6 +444,366 @@ VALUE <key> <flags> <bytes> [<cas unique>]\r\n
+- q: use noreply semantics for return codes
+- S (token): size of <data block> to store
+- T (token): Time-To-Live for item, see "Expiration" above.
+
+The flags are now repeated with detailed information where useful:
+
+- C (token): compare CAS value when storing item
+
+Similar to the basic "cas" command, only store item if the supplied token
+matches the current CAS value of the item. When combined with the 'I' flag, a
+CAS value that is _lower_ than the current value may be accepted, but the item
+will be marked as "stale", returning the X flag with mget requests.
+
+- F (token): set client flags to token (32 bit unsigned numeric)
+
+Sets flags to 0 if not supplied.
+
+- I: invalid. set-to-invalid if CAS is older than it should be.
+
+Functional when combined with 'C' flag above.
+
+- O (token): opaque value, consumes a token and copies back with response
+
+See description under 'Meta Get'
+
+- q: use noreply semantics for return codes
+
+Noreply is a method of reducing the amount of data sent back by memcached to
+the client for normal responses. In the case of metaset, a response that would
+start with "ST" (STORED) will not be sent. Any other code, such as "EX"
+(EXISTS) will still be returned.
+
+Errors are always returned.
+
+Meta Delete
+-----------
+
+The meta delete command allows for explicit deletion of items, as well as
+marking items as "stale" to allow serving items as stale during revalidation.
+
+md <key> <flags> <tokens>*\r\n
+
+- <key> means one key string.
+
+- <flags> are a set of single character codes ended with a space or newline.
+
+- <tokens>* means zero or more tokens, based on the flags supplied. They are
+  consumed in order specified by the flags.
+
+The response is in the format:
+
+CD <flags> <tokens>*\r\n
+
+Where CD is one of:
+
+- "DE" (DELETED), to indicate success
+
+- "NF" (NOT_FOUND), to indicate that the item with this key was not found.
+
+- "EX" (EXISTS), to indicate that the supplied CAS token does not match the
+  stored item.
+
+The flags used by the 'md' command are:
+
+- C (token): compare CAS value
+- I: invalidate. mark as stale, bumps CAS.
+- k: return key
+- O: opaque to copy back.
+- q: noreply
+- T (token): updates TTL
+
+The flags are now repeated with detailed information where useful:
+
+- C (token): compare CAS value
+
+Can be used to only delete or mark-stale if a supplied CAS value matches.
+
+- I: invalidate. mark as stale, bumps CAS.
+
+Instead of removing an item, this will give the item a new CAS value and mark
+it as stale. This means when it is later fetched by metaget, the client will
+be supplied an 'X' flag to show the data is stale and needs to be recached.
+
+- O (token): opaque to copy back.
+
+See description under 'Meta Get'
+
+- q: noreply
+
+See description under 'Meta Set'. In the case of meta delete, this will hide
+response lines with the code "DE". It will still return any other responses.
+
+- T (token): updates TTL
```

Comment:
May be it's good to mentioned this option is only valid with `I` for `md`

Reply:
It does below, but I'll make that more explicit.